### PR TITLE
Revert PR 32299, flag users when Zero-3 was missed

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Reverts https://github.com/huggingface/transformers/pull/32299 until I can come up with a better solution. 

TL;DR:

1. This broke quantization, which was attempted to fix in this PR: https://github.com/huggingface/transformers/pull/32640
2. However, this breaks Zero-3 inference plainly, so for now disabled while leaving context in the test. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for reverting!

**Review by ArthurZucker — APPROVED** [review]
DId any of our CI fail ? You linked a PR but not issue so wondering if we are properly testing? 🤗

**Comment by muellerzr** [comment]
None that I can *tell*, but our DS CI needs work to actually make the error logs readable. I'll see what I can do.
